R/sample_pseudoabs.R

```diff
@@ -41,11 +41,23 @@
 
 sample_pseudoabs <- function (data, raster, n, coords = NULL,
                               method="random", class_label = "pseudoabs",
-                              return_pres=TRUE)
+                              return_pres = TRUE)
 {
   return_sf <- FALSE # flag whether we need to return an sf object
-  if (inherits(data,"sf")){
-    data <- data %>% dplyr::bind_cols(sf::st_coordinates(data))
+  if (inherits(data,"sf")) {
+    bind_col <- TRUE
+    if (all(c("X", "Y") %in% names(data))) {
+      if (any(is.na(data[, c("X", "Y")]))) {
+        stop("sf object contains NA values in the X and Y coordinates")
+        } else if (all(st_drop_geometry(data[, c("X", "Y")]) == sf::st_coordinates(data))) {
```

Comment:
Overall, the changes deal with all the issues that we discussed. But the current code fails Rcheck. From a quick look, there is a problem with `st_drop_geometry` being used with properly calling the package (i.e. sf::st_drop_geometry). Can you please fix it, and make sure that your pull request passes all checks? We are almost there. Thanks.